gprofiler/utils.py

```diff
@@ -97,12 +98,27 @@ def start_process(cmd: Union[str, List[str]], via_staticx: bool, **kwargs) -> Po
             env = env if env is not None else os.environ.copy()
             env.update({"LD_LIBRARY_PATH": ""})
 
+    cur_preexec_fn = kwargs.pop("preexec_fn", os.setpgrp)
+
+    if kill_on_parent_death:
+        def wrap_with_set_pseg_death(inner):
+            def wrapper():
+                PR_SET_PDEATHSIG = 1
+                TERM = 15
+                libc.prctl(PR_SET_PDEATHSIG, TERM)
+                if inner:
+                    return inner()
+
+            return wrapper
```

Comment:
Please extract the prctl function to some utils section.

Reply:
done